great_expectations/checkpoint/actions.py

```diff
@@ -179,42 +236,44 @@ class SlackNotificationAction(DataDocsAction):
         show_failed_expectations: Shows a list of failed expectation types.
     """
 
-    def __init__(  # noqa: PLR0913
-        self,
-        renderer: dict,
-        slack_webhook: Optional[str] = None,
-        slack_token: Optional[str] = None,
-        slack_channel: Optional[str] = None,
-        notify_on: str = "all",
-        notify_with: Optional[list[str]] = None,
-        show_failed_expectations: bool = False,
-    ) -> None:
-        """Create a SlackNotificationAction"""
-        super().__init__()
-        self.renderer = instantiate_class_from_config(
-            config=renderer,
-            runtime_environment={},
-            config_defaults={},
-        )
-        module_name = renderer["module_name"]
-        if not self.renderer:
-            raise ClassInstantiationError(
-                module_name=module_name,
-                package_name=None,
-                class_name=renderer["class_name"],
+    class Config:
+        arbitrary_types_allowed = True
+        json_encoders = {SlackRenderer: lambda s: _encode_renderer(s)}
```

Comment:
Decided to use rich types in the `__init__` but write class_name/module_name to disk when serializing